python/ray/train/callbacks/logging.py

```diff
@@ -63,24 +66,10 @@ def __init__(self,
                  workers_to_log: Optional[Union[int, List[int]]] = 0) -> None:
```

Comment:
Might be out of scope for this PR, but taking a second look at the logging callback structure, I think it makes more sense for `TrainingLogdirCallback` and `TrainingSingleFileLoggingCallback` to be mixins or utilities instead of callbacks themselves. Each of them only are used to create a logdir or a logging file, and don't need to know about results or preprocessors.

Reply:
Yep agreed, now that there's been some good general discussion/context in this PR let me spin this off into 2 PRs:
1. General Callback structure (including logging) 
2. PyTorch Profiling